<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CustomersWidget</class>
 <widget class="QWidget" name="CustomersWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Gestión de Clientes</string>
  </property>
  <layout class="QVBoxLayout" name="layoutPrincipal">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>10</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>10</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   
   <!-- Header -->
   <item>
    <widget class="QFrame" name="frameHeader">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QHBoxLayout" name="layoutHeader">
      <item>
       <widget class="QLabel" name="lblTitulo">
        <property name="text">
         <string>👥 Gestión de Clientes</string>
        </property>
        <property name="styleSheet">
         <string>font-size: 18px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="spacerHeader">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnNuevoCliente">
        <property name="text">
         <string>➕ Nuevo Cliente</string>
        </property>
        <property name="toolTip">
         <string>Crear nuevo cliente (Ctrl+N)</string>
        </property>
        <property name="styleSheet">
         <string>QPushButton { background-color: #27ae60; }</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   
   <!-- Filtros y Búsqueda -->
   <item>
    <widget class="QGroupBox" name="groupFiltros">
     <property name="title">
      <string>Filtros y Búsqueda</string>
     </property>
     <layout class="QHBoxLayout" name="layoutFiltros">
      <item>
       <widget class="QLabel" name="lblBuscar">
        <property name="text">
         <string>Buscar:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEditBuscar">
        <property name="placeholderText">
         <string>Nombre, código, email o teléfono...</string>
        </property>
        <property name="toolTip">
         <string>Buscar cliente por cualquier campo</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblSegmento">
        <property name="text">
         <string>Segmento:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cmbSegmento">
        <property name="toolTip">
         <string>Filtrar por segmento de cliente</string>
        </property>
        <item>
         <property name="text">
          <string>Todos</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>REGULAR</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>VIP</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>PREMIUM</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblEstado">
        <property name="text">
         <string>Estado:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cmbEstado">
        <property name="toolTip">
         <string>Filtrar por estado del cliente</string>
        </property>
        <item>
         <property name="text">
          <string>Todos</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ACTIVO</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>INACTIVO</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>POTENCIAL</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>PERDIDO</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnLimpiarFiltros">
        <property name="text">
         <string>🧹 Limpiar</string>
        </property>
        <property name="toolTip">
         <string>Limpiar todos los filtros</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   
   <!-- Contenido Principal -->
   <item>
    <layout class="QHBoxLayout" name="layoutPrincipal">
     
     <!-- Lista de Clientes -->
     <item>
      <widget class="QGroupBox" name="groupClientes">
       <property name="title">
        <string>Lista de Clientes</string>
       </property>
       <layout class="QVBoxLayout" name="layoutClientes">
        <item>
         <widget class="QTableWidget" name="tblClientes">
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Código</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Nombre</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Email</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Teléfono</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Segmento</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Estado</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Total Compras</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Última Compra</string>
           </property>
          </column>
         </widget>
        </item>
        
        <!-- Acciones de Tabla -->
        <item>
         <layout class="QHBoxLayout" name="layoutAccionesTabla">
          <item>
           <widget class="QLabel" name="lblTotalClientes">
            <property name="text">
             <string>Total: 0 clientes</string>
            </property>
            <property name="styleSheet">
             <string>font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="spacerAcciones">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btnExportarClientes">
            <property name="text">
             <string>📊 Exportar</string>
            </property>
            <property name="toolTip">
             <string>Exportar lista de clientes</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnImportarClientes">
            <property name="text">
             <string>📥 Importar</string>
            </property>
            <property name="toolTip">
             <string>Importar clientes desde archivo</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     
     <!-- Panel de Detalles -->
     <item>
      <widget class="QGroupBox" name="groupDetalles">
       <property name="title">
        <string>Detalles del Cliente</string>
       </property>
       <property name="maximumWidth">
        <number>350</number>
       </property>
       <layout class="QVBoxLayout" name="layoutDetalles">
        
        <!-- Información Básica -->
        <item>
         <widget class="QGroupBox" name="groupInfoBasica">
          <property name="title">
           <string>Información Básica</string>
          </property>
          <layout class="QFormLayout" name="layoutInfoBasica">
           <item row="0" column="0">
            <widget class="QLabel" name="lblDetalleNombre">
             <property name="text">
              <string>Nombre:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="lblDetalleNombreValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblDetalleCodigo">
             <property name="text">
              <string>Código:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="lblDetalleCodigoValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblDetalleEmail">
             <property name="text">
              <string>Email:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="lblDetalleEmailValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblDetalleTelefono">
             <property name="text">
              <string>Teléfono:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="lblDetalleTelefonoValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        
        <!-- Estadísticas -->
        <item>
         <widget class="QGroupBox" name="groupEstadisticas">
          <property name="title">
           <string>Estadísticas</string>
          </property>
          <layout class="QFormLayout" name="layoutEstadisticas">
           <item row="0" column="0">
            <widget class="QLabel" name="lblDetalleSegmento">
             <property name="text">
              <string>Segmento:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="lblDetalleSegmentoValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblDetalleEstado">
             <property name="text">
              <string>Estado:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="lblDetalleEstadoValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblDetalleTotalCompras">
             <property name="text">
              <string>Total Compras:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="lblDetalleTotalComprasValor">
             <property name="text">
              <string>$0.00</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold; color: #27ae60;</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblDetalleNumCompras">
             <property name="text">
              <string>Núm. Compras:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="lblDetalleNumComprasValor">
             <property name="text">
              <string>0</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="lblDetalleUltimaCompra">
             <property name="text">
              <string>Última Compra:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="lblDetalleUltimaCompraValor">
             <property name="text">
              <string>-</string>
             </property>
             <property name="styleSheet">
              <string>font-weight: bold;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        
        <!-- Acciones -->
        <item>
         <layout class="QVBoxLayout" name="layoutAccionesCliente">
          <item>
           <widget class="QPushButton" name="btnEditarCliente">
            <property name="text">
             <string>✏️ Editar Cliente</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Editar cliente seleccionado</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnEliminarCliente">
            <property name="text">
             <string>🗑 Eliminar Cliente</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string>QPushButton { background-color: #e74c3c; }</string>
            </property>
            <property name="toolTip">
             <string>Eliminar cliente seleccionado</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnHistorialCompras">
            <property name="text">
             <string>📊 Ver Historial</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Ver historial de compras del cliente</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnAnalisisPredictivo">
            <property name="text">
             <string>🔮 Análisis Predictivo</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Ver análisis predictivo del cliente</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        
        <!-- Spacer -->
        <item>
         <spacer name="spacerDetalles">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </spacer>
        </item>
        
       </layout>
      </widget>
     </item>
     
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>